### PR TITLE
Seed extractor now checks for variant when vending seeds from the extractor

### PR DESCRIPTION
**Pull request by Migratingcocofruit**
## What Does This PR Do
Previously seed extractors would vend seeds based only on their names and stats, without taking variant into account. This could result in a different seed being vended from the one selected.

## Why It's Good For The Game
Seed extractor no longer vends the wrong seed. 

## Images of changes
Before:

https://github.com/ParadiseSS13/Paradise/assets/69551563/de8e5197-4a84-4935-8659-8006ee8808b3

After:

https://github.com/ParadiseSS13/Paradise/assets/69551563/02624495-d9a9-4eb3-a9fc-e501eada08e8

## Testing
Made a bunch of seed variants, inserted and vended them.

## Changelog
:cl:
fix: Seed extractor now vends seeds of the correct variant
/:cl:

### CODE REVIEW

File: code/modules/hydroponics/seed_extractor.dm [hunk 1]
```diff
@@ -157,7 +157,7 @@
 	. = FALSE
 	switch(action)
 		if("vend")
-			vend_seed(text2num(params["seedid"]), vend_amount)
+			vend_seed(text2num(params["seedid"]),params["seedvariant"], vend_amount)
```
Comment (Burzah) [inline code/modules/hydroponics/seed_extractor.dm:160]:
```suggestion
			vend_seed(text2num(params["seedid"]), params["seedvariant"], vend_amount)
```

Comment (Burzah) [inline code/modules/hydroponics/seed_extractor.dm:160]:
```suggestion
			vend_seed(text2num(params["seedid"]), params["seedvariant"], vend_amount)
```

**Review by S34NW — APPROVED** [review]
deconflict for merge

**Comment by Migratingcocofruit** [comment]
> deconflict for merge

Will do next time I'm home. Not sure when.

**Pull request merged**